pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -484,12 +484,25 @@ func (vs *vSphereVMProvider) updateVirtualMachine(
 func (vs *vSphereVMProvider) vmCreateDoPlacement(
 	vmCtx pkgctx.VirtualMachineContext,
 	vcClient *vcclient.Client,
-	createArgs *VMCreateArgs) error {
+	createArgs *VMCreateArgs) (retErr error) {
+
+	defer func() {
```

Comment:
Looking at the code. It does seem like we could have set the condition all along for this error. Since the result returned was not sufficient for IS.
Maybe @bryanv has a reasoning behind why it wasn't surfaced earlier?.